### PR TITLE
[LoRA] add: test to check if peft loras are loadable in non-peft envs.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Adds a test to ensure LoRA trained using `peft` (emulating how it's done in our trainers) can be fully loaded without `peft`. 

Helps to keep our sanity in check :D 

The dummy components were obtained using this Colab:
https://huggingface.co/datasets/diffusers/notebooks/blob/main/check_logits_with_serialization_peft_lora.py. This script also spits out the logits which are tested in this PR. 

The basic idea is the following. 

If we do:

```python
sd_pipeline = DiffusionPipeline.from_pretrained(...)

torch.manual_seed(0)
unet_lora_config = LoraConfig(...)

torch.manual_seed(0)
text_encoder_lora_config = LoraConfig(...)

sd_pipeline.unet.add_adapter(unet_lora_config)
sd_pipeline.text_encoder.add_adapter(unet_lora_config)

outputs = sd_pipeline(**pipeline_inputs, generator=torch.manual_seed(0), output_type="np").images
```

And then if you do (with `peft` uninstalled):

```python
sd_pipeline = DiffusionPipeline.from_pretrained(...)

# LoRA was obtained using what's shown in the script linked above. 
sd_pipeline.load_lora_weights(...)
outputs_no_peft = sd_pipeline(**pipeline_inputs, generator=torch.manual_seed(0), output_type="np").images
```

`outputs` and `outputs_no_peft` should match. This is exactly what we test here. 

Cc: @apolinario

### CODE REVIEW

**Comment by sayakpaul** [comment]
@younesbelkada just did. PTAL.

**Review by younesbelkada — APPROVED** [review]
Awesome work, thanks a lot @sayakpaul ! 🚀

**Comment by sayakpaul** [comment]
@younesbelkada feel free to look at the Colab with which I generated the LoRA files: https://colab.research.google.com/gist/sayakpaul/4a00d0223c03225f82735ff93930f43d/scratchpad.ipynb

I think they emulate how it's done in the training scripts. But I'd appreciate another set of 👁️s.

**Comment by DN6** [comment]
I think we might need a check to ensure this doesn't run in the PEFT environment as well right?

**Comment by sayakpaul** [comment]
Why can't it run on `peft` environment? Could you elaborate?

**Comment by DN6** [comment]
Why run the test in both the PEFT and non-PEFT environment/runners if it is only meant to check that we can load PEFT LoRAs in a non-PEFT environment?

**Comment by sayakpaul** [comment]
Oh okay. I think we're covering on that front:

https://github.com/huggingface/diffusers/blob/17546020fc0e7a09e367540ac578e26e49a2e8be/.github/workflows/pr_tests.yml#L100 doesn't run PEFT LoRA stuff. 

All tests from https://github.com/huggingface/diffusers/blob/main/tests/lora/test_lora_layers_peft.p (which is tested in [this workflow](https://github.com/huggingface/diffusers/blob/17546020fc0e7a09e367540ac578e26e49a2e8be/.github/workflows/pr_test_peft_backend.yml#L61)), always requires the PEFT backend:

https://github.com/huggingface/diffusers/blob/17546020fc0e7a09e367540ac578e26e49a2e8be/tests/lora/test_lora_layers_peft.py#L78

Does this make sense?

**Review by patrickvonplaten — APPROVED** [review]
Good idea!
